client/my-sites/site-settings/wrap-settings-form.jsx

```diff
@@ -138,16 +155,32 @@ const wrapSettingsForm = getFormSettings => SettingsForm => {
 	const connectComponent = connect(
 		state => {
 			const siteId = getSelectedSiteId( state );
-			const isRequestingSettings = isRequestingSiteSettings( state, siteId );
-			const isSavingSettings = isSavingSiteSettings( state, siteId );
-			const isSaveRequestSuccessful = isSiteSettingsSaveSuccessful( state, siteId );
-			const settings = getSiteSettings( state, siteId );
+			let isRequestingSettings = isRequestingSiteSettings( state, siteId );
+			let isSavingSettings = isSavingSiteSettings( state, siteId );
+			let isSaveRequestSuccessful = isSiteSettingsSaveSuccessful( state, siteId );
+			let settings = getSiteSettings( state, siteId );
+			const settingsFields = {
+				site: keys( settings ),
+			};
+
+			const isJetpack = isJetpackSite( state, siteId );
+			if ( isJetpack ) {
+				const jetpackSettings = getJetpackSettings( state, siteId );
+				isRequestingSettings = isRequestingSettings || isRequestingJetpackSettings( state, siteId );
+				isSavingSettings = isSavingSettings || isUpdatingJetpackSettings( state, siteId );
+				isSaveRequestSuccessful = isSaveRequestSuccessful && isJetpackSettingsSaveSuccessful( state, siteId );
+				settings = { ...settings, ...jetpackSettings };
+				settingsFields.jetpack = keys( jetpackSettings );
+			}
+
 			return {
-				isRequestingSettings: isRequestingSettings && ! settings,
+				isRequestingSettings,
```

Comment:
This introduces a difference compared to the old behaviour where the form was not disabled if we had cached settings, in this case, each time we show a form, we'll have to wait for the fetch settings to complete before being able to edit the settings even if we had cached the settings before.

could we make the flag falsy if the settings (and jetpackSettings for jetpack sites) are available?